tests/faker/conversation_service.py

```diff
@@ -0,0 +1,165 @@
+from tests.faker.conversation import Conversation
+from tests.faker.events import (
+    Event,
+    RegistrationsEvent,
+    ErrorsEvent,
+    TransferCompatibilityStatusesEvent,
+    EhrRequestsEvent,
+    EhrResponsesEvent,
+    EhrIntegrationsEvent,
+    ReadyToIntegrateStatusesEvent,
+)
+from tests.test_base import EventType
+
+import json
+from uuid import uuid4
+
+
+class ConversationService:
+    def generate_conversations(
+        self,
+        number_of_conversations: int,
+        number_to_error_at_registrations: int,
+        number_to_error_at_transfer_compatibility_statuses: int,
+        number_to_error_at_ehr_request: int,
+        number_to_error_at_ehr_response: int,
+        number_to_error_at_ready_to_integrate: int,
+        number_to_error_at_integration: int,
+    ):
+        conversations = []
+
+        for i in range(number_of_conversations):
+            # create a new conversation with generates a conversationId and a list to hold all the events for that conversation.
+            conversation = Conversation()
+
+            # add to conversations collection
+            conversations.append(conversation)
+
+            # create the first event - REGISTRATION
+            (
+                reg_event,
+                reg_error_event,
+                number_to_error_at_registrations,
+            ) = self._generate_events(
+                EventType.REGISTRATIONS, conversation, number_to_error_at_registrations
+            )
+
+            if not reg_error_event is None:
```

Comment:
Todo: you can writer this as `if reg_error_event is not None:` and this way is clearer. Need to change this throughout.